src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,440 @@
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace;
+
+import java.util.Locale;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Checks for empty line separators at the beginning/end of a code block.
+ *
+ * <p> The policy to verify is specified using the {@link BlockWrapOption} class and
+ * defaults to {@link BlockWrapOption#EMPTY_LINE_ALLOWED}. Policies for the beginning and
+ * the end of the block are configured separately.
+ * </p>
+ *
+ * <p> The check is for any statement that have a body:
+ * {@link TokenTypes#ANNOTATION_ARRAY_INIT ANNOTATION_ARRAY_INIT},
+ * {@link TokenTypes#ANNOTATION_DEF ANNOTATION_DEF},
+ * {@link TokenTypes#ARRAY_INIT ARRAY_INIT},
+ * {@link TokenTypes#CASE_GROUP CASE_GROUP},
+ * {@link TokenTypes#CLASS_DEF CLASS_DEF},
+ * {@link TokenTypes#CTOR_DEF CTOR_DEF},
+ * {@link TokenTypes#ENUM_CONSTANT_DEF ENUM_CONSTANT_DEF},
+ * {@link TokenTypes#ENUM_DEF ENUM_DEF},
+ * {@link TokenTypes#INSTANCE_INIT INSTANCE_INIT},
+ * {@link TokenTypes#INTERFACE_DEF INTERFACE_DEF},
+ * {@link TokenTypes#LAMBDA LAMBDA},
+ * {@link TokenTypes#LITERAL_CASE LITERAL_CASE},
+ * {@link TokenTypes#LITERAL_CATCH LITERAL_CATCH},
+ * {@link TokenTypes#LITERAL_DEFAULT LITERAL_DEFAULT},
+ * {@link TokenTypes#LITERAL_DO LITERAL_DO},
+ * {@link TokenTypes#LITERAL_ELSE LITERAL_ELSE},
+ * {@link TokenTypes#LITERAL_FINALLY LITERAL_FINALLY},
+ * {@link TokenTypes#LITERAL_FOR LITERAL_FOR},
+ * {@link TokenTypes#LITERAL_IF LITERAL_IF},
+ * {@link TokenTypes#LITERAL_NEW LITERAL_NEW},
+ * {@link TokenTypes#LITERAL_SWITCH LITERAL_SWITCH},
+ * {@link TokenTypes#LITERAL_SYNCHRONIZED LITERAL_SYNCHRONIZED},
+ * {@link TokenTypes#LITERAL_TRY LITERAL_TRY},
+ * {@link TokenTypes#LITERAL_WHILE LITERAL_WHILE},
+ * {@link TokenTypes#METHOD_DEF METHOD_DEF},
+ * {@link TokenTypes#OBJBLOCK OBJBLOCK},
+ * {@link TokenTypes#SLIST SLIST},
+ * {@link TokenTypes#STATIC_INIT STATIC_INIT}.
+ * </p>
+ *
+ * <p> By default the check will check the following statements:
+ *  {@link TokenTypes#ANNOTATION_DEF ANNOTATION_DEF},
+ *  {@link TokenTypes#CLASS_DEF CLASS_DEF},
+ *  {@link TokenTypes#ENUM_DEF ENUM_DEF},
+ *  {@link TokenTypes#INTERFACE_DEF INTERFACE_DEF}.
+ * </p>
+ *
+ * <p> An example of how to configure the check with default parameters is:
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="EmptyLineWrappingInBlock"/&gt;
+ * </pre>
+ *
+ * <p> Example of declarations with empty line separator at the top of the block
+ * that is expected by the Check by default:
```

Comment:
> expected by the Check by default

should be  "expected by the Check by default (no violations)".

Reply:
done.